Individual.cpp

```diff
@@ -25,10 +25,55 @@ Individual::Individual(const Graph &graph) {
     generate_neighbour_vector(graph);
 }
 
+// create an offspring from two parents i.e. perform crossover between two parents
+// here, a binary mask of length equal to the number of nodes is randomly
+// created, and an offspring is generated by selecting from the
+// first parent the genes where the mask is 0, and from the second
+// parent the genes where the mask is 1.
+Individual::Individual(const Individual &individual1, const Individual &individual2) {
+    int no_chromosomes = individual1.size();
+
+    for (int i = 0; i < no_chromosomes; i++) {
+        int bit = rand() % 2;
```

Comment:
It's faster if you use (rand() & 1) to get the last bit of the random number, which is the same as getting the parity of that number.